### PR TITLE
Java Program to check if a singly linked list is a palindrome or not.

### PR DESCRIPTION
**Pull request by salma2vec**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes, I have.
### Description

(Write your answer here.)
This is a Java Program to check if a singly linked list is a palindrome or not. I have included proper comments throughout the code, whenever found necessary.
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Few changes: 
- Make markdown updates
- Add comments at proper places
- Add a beginner description about the code
- Add dynamic input
- Add sample I/O and complexity at the end of the code

**Comment by salma2vec** [comment]
@HarshCasper I have made the required changes. And, I don't get the one failing check - Java(beta)...

**Comment by HarshCasper** [comment]
> @HarshCasper I have made the required changes. And, I don't get the one failing check - Java(beta)...

No you have not. Make the README (Markdown) changes.

**Comment by IdealisticINTJ** [comment]
I made the README changes. I am sorry. I forgot to, earlier.

**Comment by salma2vec** [comment]
@HarshCasper I have made the changes 2 days ago. Could you please check it?

File: Java/other/PalindromeLinkedList.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
+// This program is to check if a singly linked list is a palindrome or not.
+// And, A singly linked list is a linked list that can be traversed in only one direction from head to the last node, i.e Unidirectional essentially.
+// A string is said to be palindrome if it reads the same backward as forward. 
+// eg. racecar, madam etc.
```
Comment (HarshCasper) [inline Java/other/PalindromeLinkedList.java]: Use Multi-Line Comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by atarax665 — APPROVED** [review]
LGTM 👍

**Comment by atarax665** [comment]
Do some minor suggested changes and resolve the conflict so that I could merge the PR.

**Comment by salma2vec** [comment]
Did! @atarax665

**Comment by salma2vec** [comment]
I made the changes 3 days ago. Please review it @HarshCasper @atarax665.

**Comment by atarax665** [comment]
I can still see the merge conflict, please do resolve it.

**Review by ankitaggarwal23 — COMMENTED** [review]
Code plagiarized from [here.](https://www.javatpoint.com/java-program-to-determine-whether-a-singly-linked-list-is-the-palindrome)

**Pull request closed without merging**